bin/check_smoker_plugin.py

```diff
@@ -22,51 +26,94 @@
 
 parser = argparse.ArgumentParser(description='Smoker Nagios plugin')
 parser.add_argument('-f', '--force', dest='force', action='store_true', help="Force plugins run (otherwise just print last results)")
-parser.add_argument('plugin', help="Plugin to check")
+parser.add_argument('--category', default=None, help='The category of tests to run')
+parser.add_argument('--component', default=None, help='The tests component to run')
+parser.add_argument('--health', default=False, dest='health', action='store_true', help="Run only healthchecks")
+parser.add_argument('plugin', default=None, nargs='*', help="Plugin to check")
 args = parser.parse_args()
 
-def ParseIssue(plugin_result):
+
+def parse_issue(plugin_result):
     out = "smoke test: %s " % plugin_result['name']
     for component in plugin_result['lastResult']['componentResults']:
-        if component['componentResult']['status'] != 'OK':
-            out += "%s: %s; " % ( component['componentResult']['name'], component['componentResult']['status'] )
+        if component['componentResult']['status'] != OK:
+            out += "%s: %s; " % ( component['componentResult']['name'], component['componentResult']['status'])
     if len(out) > 1024:
         out = "smoke test: %s returns huge error output, see smoker logs for details" % plugin_result['name']
     return out
 
+
 def main():
     client = Client(['localhost'])
-    plugins = client.get_plugins([[args.plugin]])
+    plugins = None
+
+    if args.plugin:
+        if args.category or args.component or args.health:
+            lg.warn("Plugins specified by name, ignoring --category, --component and --health")
+
+        plugins = client.get_plugins([args.plugin])
+    elif args.category or args.component:
+        filter = []
+        if args.category:
+            filter.append({'key': 'Category', 'value': args.category})
+        if args.component:
+            filter.append({'key': 'Component', 'value': args.component})
+        if args.health:
+            filter.append({'key': 'Type', 'value': 'healthCheck'})
+        plugins = client.get_plugins(filter)
+    else:
+        nagios.exit_unknown("invalid startup configuration - neither plugin nor --category nor --component specified")
 
     # No plugin found
     if not plugins:
-        nagios.exit_unknown("plugin %s not found" % args.plugin)
+        if args.plugin:
+            nagios.exit_unknown("plugin %s not found" % args.plugin)
+        else:
+            nagios.exit_ok("no plugin found by category %s and component %s, health: %s"
```

Comment:
This is actually by design. The ultimate incentive for modifications of check_smoker_plugin is to run Eureka healthchecks. However, Eureka is not yet deployed in some environments (particularly na1 and eu). For some time therefore, this check will not actually run any smoker plugins and it would not be comfortable to keep seeing warnings. Once Eureka is deployed and tested everywhere, we can change the behavior to what you suggest.